common/src/main/java/com/microsoft/identity/common/adal/internal/cache/StorageHelper.java

```diff
@@ -225,76 +246,136 @@ public String encrypt(final String clearText)
 
         final String encryptedText = new String(Base64.encode(blobVerAndEncryptedDataAndIVAndMacDigest,
                 Base64.NO_WRAP), AuthenticationConstants.ENCODING_UTF8);
-        Log.v(TAG, "Finished encryption");
+        Logger.verbose(TAG + methodName, "Finished encryption");
 
         return getEncodeVersionLengthPrefix() + ENCODE_VERSION + encryptedText;
     }
 
     @Override
     public String decrypt(final String encryptedBlob) throws GeneralSecurityException, IOException {
         final String methodName = "decrypt";
-        Log.v(TAG, "Starting decryption");
+        Logger.verbose(TAG + methodName, "Starting decryption");
 
         if (StringExtensions.isNullOrBlank(encryptedBlob)) {
             throw new IllegalArgumentException("Input is empty or null");
         }
 
-        int encodeVersionLength = encryptedBlob.charAt(0) - 'a';
-        validateEncodeVersion(encryptedBlob, encodeVersionLength);
+        if (getEncryptionType(encryptedBlob) == EncryptionType.UNENCRYPTED) {
+            Logger.verbose(TAG + methodName, "This string is not encrypted. Finished decryption.");
+            return encryptedBlob;
+        }
 
-        final byte[] bytes = Base64.decode(
-                encryptedBlob.substring(1 + encodeVersionLength),
-                Base64.DEFAULT
-        );
+        final String packageName = getPackageName();
+        final List<KeyType> potentialDecryptionKeyList = initializeDecryptionKeyTypeList(encryptedBlob, packageName);
+
+        final byte[] bytes = getByteArrayFromEncryptedBlob(encryptedBlob);
+        for (final KeyType keyType : potentialDecryptionKeyList) {
+            try {
+                final SecretKey secretKey = loadSecretKey(keyType);
+                if (secretKey == null) {
+                    continue;
+                }
+
+                String result = decryptWithSecretKey(bytes, secretKey);
+                Logger.verbose(TAG + methodName, "Finished decryption.");
+                return result;
+            } catch (GeneralSecurityException | IOException e) {
+                Logger.error(
+                        TAG + methodName,
+                        "Failed to decrypt with KeyType: "
+                                + keyType.toString(),
+                        e
+                );
+            }
+        }
 
-        final String packageName = mContext.getPackageName();
+        Logger.verbose(
+                TAG + methodName,
+                "Tried all decryption keys and decryption still fails. Throw an exception.");
+
+        throw new GeneralSecurityException(ErrorStrings.DECRYPTION_FAILED);
+    }
+
+    /**
+     * Determine type of encryption performed on the given data blob.
+     * NOTE :If it cannot verify the keyVersion, it will assume that this data is not encrypted.
+     * */
+    public EncryptionType getEncryptionType(@NonNull final String data) throws IOException {
+        final String methodName = "isEncryptedWithUserDefinedKey";
```

Comment:
The method name string has colon as prefix in some methods whereas it is not present in other places such as this one. I think we should adopt one pattern and be consistent. For example, we can change this to `final String methodName = ":isEncryptedWithUserDefinedKey";` and do the same in all other places as well.